core/src/main/java/tc/oc/pgm/variables/VariableDefinition.java

```diff
@@ -0,0 +1,30 @@
+package tc.oc.pgm.variables;
+
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.features.SelfIdentifyingFeatureDefinition;
+import tc.oc.pgm.filters.Filterable;
+
+public class VariableDefinition<T extends Filterable<?>> extends SelfIdentifyingFeatureDefinition {
+
+  private final Class<T> scope;
+  private final double def;
+
+  public VariableDefinition(String id, Class<T> scope, double def) {
+    super(id);
+    this.scope = scope;
+    this.def = def;
+  }
+
+  public Class<T> getScope() {
+    return scope;
+  }
+
+  public double getDefault() {
+    return def;
+  }
+
+  public Variable<T> getVariable(Match match) {
+    //noinspection unchecked
+    return (Variable<T>) match.getFeatureContext().get(this.getId());
```

Comment:
I still think this is a weird and unnecessary deviation from the current PGM norms of suppressing warnings. I see no reason to change it?

```suggestion
  @SupressWarnings("unchecked")
  public Variable<T> getVariable(Match match) {
    return (Variable<T>) match.getFeatureContext().get(this.getId());
```

Reply:
Sure, can move that to the method scope